src/annotator/cross-frame.js

```diff
@@ -59,71 +59,88 @@ export class CrossFrame {
+  }
 
-    /**
-     * Remove the connection between the sidebar and annotator.
-     */
-    this.destroy = () => {
-      bridge.destroy();
-      annotationSync.destroy();
-      discovery.stopDiscovery();
-      frameObserver.disconnect();
-    };
+  /**
+   * Remove the connection between the sidebar and annotator.
+   */
+  destroy() {
+    this.bridge.destroy();
+    this.annotationSync.destroy();
+    this.discovery.stopDiscovery();
+    this.frameObserver.disconnect();
+  }
 
-    /**
-     * Notify the sidebar about new annotations created in the page.
-     *
-     * @param {AnnotationData[]} annotations
-     */
-    this.sync = annotations => annotationSync.sync(annotations);
+  /**
+   * Notify the sidebar about new annotations created in the page.
+   *
+   * @param {AnnotationData[]} annotations
+   */
+  sync(annotations) {
+    this.annotationSync.sync(annotations);
+  }
 
-    /**
-     * Subscribe to an event from the sidebar.
-     *
-     * @param {string} event
-     * @param {(...args: any[]) => void} callback
-     */
-    this.on = (event, callback) => bridge.on(event, callback);
+  /**
+   * Subscribe to an event from the sidebar.
+   *
+   * @param {string} method
+   * @param {(...args: any[]) => void} listener -- Final argument is an optional
+   *   callback of the type: `(error: string|Error|null, ...result: any[]) => void`.
+   *   This callback must be invoked in order to respond (via `postMessage`)
+   *   to the other frame/s with a result or an error.
+   * @throws {Error} If trying to register a callback after a channel has already been created
+   * @throws {Error} If trying to register a callback with the same name multiple times
+   */
+  on(method, listener) {
+    this.bridge.on(method, listener);
+  }
 
-    /**
-     * Call an RPC method exposed by the sidebar to the annotator.
-     *
-     * @param {string} method
-     * @param {any[]} args
-     */
-    this.call = (method, ...args) => bridge.call(method, ...args);
+  /**
+   * Call an RPC method exposed by the sidebar to the annotator.
+   *
+   * @param {string} method
+   * @param {any[]} args - Arguments to method. Final argument is an optional
+   *   callback with this type: `(error: string|Error|null, ...result: any[]) => void`.
+   *   This callback, if any, will be triggered once a response (via `postMessage`)
+   *   comes back from the other frame/s. If the first argument (error) is `null`
+   *   it means successful execution of the whole remote procedure call.
+   */
+  call(method, ...args) {
+    this.bridge.call(method, ...args);
+  }
 
-    /**
-     * Register a callback to be invoked once the connection to the sidebar
-     * is set up.
-     *
-     * @param {(...args: any[]) => void} callback
-     */
-    this.onConnect = callback => bridge.onConnect(callback);
+  /**
+   * Register a callback to be invoked once the connection to the sidebar
+   * is set up.
+   *
+   * @param {Parameters<Bridge['onConnect']>[0]} callback
```

Comment:
I preferred just duplicating the original type here (`(...args: any[]) => void`). This is easier for readers to see what the type is at a glance than mentally unpack the `Parameters<...>[0]` expression. In general I would tend to only use things like `Parameters<...>`/`ReturnType<...>` where the expression is dynamic or very complex.

Reply:
Agree.